igvc_utils/include/mocking_utils/mock_subscriber.h

```diff
@@ -0,0 +1,149 @@
+//
+// Created by vivek on 3/1/20.
+//
+
+#include <gtest/gtest.h>
+#include <igvc_msgs/velocity_pair.h>
+#include <ros/callback_queue.h>
+#include <ros/ros.h>
+#include <sensor_msgs/Joy.h>
+
+/**
+ * \brief A wrapper class around a Subscriber for testing.
+ *
+ * A MockMotorSubscriber contains helper methods such as blocking until there is a publisher
+ * and spinning until a message arrives.
+ *
+ * A user provided callback function can be called to perform additional testing logic on a callback
+ *
+ * Can only use one type of message. Should be able to use any type of message.
+ */
+
+template <typename M>
+class MockMotorSubscriber
```

Comment:
This should be called `MockSubscriber` instead of `MockMotorSubscriber` as it doesn't have to be related to a motor in any way